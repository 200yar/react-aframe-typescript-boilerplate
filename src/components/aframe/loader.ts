export const loaderComponentConfig =  {

  schema: {'default': ''},

  /**
   * Do not use arrow function here
   * "this: any" is needed for the typescript compiler
   * https://github.com/Microsoft/TypeScript/issues/16016#issuecomment-303462193
   */
  init: function(this: any) {
    this.el.addEventListener('loaded', () => {
      console.log('scene loaded');
      const loader = document.getElementById('loader') as HTMLDivElement;
      (loader.parentElement as HTMLDivElement).removeChild(loader);
    })
  }
}
