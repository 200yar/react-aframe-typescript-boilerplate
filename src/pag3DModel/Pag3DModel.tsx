//todo: sonido inicial
//todo: reportar bug en aframe-extras repo

import * as React from 'react';
import 'aframe-extras/dist/aframe-extras.loaders.min'; // for json loader
import Loader from "../components/loader/LoaderCmp";
import SideMenu from "../components/sideMenu/SideMenuCmp";
import TopMenu from "../components/topMenu/TopMenuCmp";
import {SIDE_MENU_ITEMS} from "../components/sideMenu/SideMenuItems";

enum BotStatus {IDLE = 'idle', RUN = 'run', WALK = 'walk', JUMP = 'jump'}

export class Pag3DModel extends React.Component<{}, {botStatus: BotStatus}> {

  public state = {botStatus: BotStatus.IDLE};
  public refs: {sideMenu: SideMenu, loader: Loader, bot: AFrame.Entity};

  // Delay between bot animation transitions in seconds. For example: walk -> run
  private crossFadeDuration = 0.5;

  private orbitControls = {
    autoRotate: true,
    target: '#bot',
    enableDamping: true,
    dampingFactor: 0.1,
    rotateSpeed: 0.1,
    autoRotateSpeed: 0.15,
    zoomSpeed: 0.5,
    minDistance: 0,
    maxDistance: 100,
    invertZoom: true
  };

  public componentDidMount() {
    this.refs.loader.hideWhen(this.refs.bot, 'model-loaded');
  }

  private onClickBtnRun = () => {
    this.setState({botStatus: BotStatus.RUN});
  };

  private onClickBtnIdle = () => {
    this.setState({botStatus: BotStatus.IDLE})
  };

  private onClickBtnWalk = () => {
    this.setState({botStatus: BotStatus.WALK})
  };

  private onClickBtnJump = () => {
    this.setState({botStatus: BotStatus.JUMP})
  };

  private openSideMenu = () => {
    this.refs.sideMenu.show();
  };

  public render() {
    return (
      <div>

        <Loader ref="loader">Loading</Loader>

        <SideMenu ref="sideMenu" title="3D Model Animation" items={ SIDE_MENU_ITEMS } itemActive="3" />

        <TopMenu onClickMenuBtn={ this.openSideMenu }>
          <a id="btnIdle" onClick={ this.onClickBtnWalk } className="top-menu-item">Walk</a>
          <a id="btnRun" onClick={ this.onClickBtnRun } className="top-menu-item">Run</a>
          <a id="btnIdle" onClick={ this.onClickBtnJump } className="top-menu-item">Jump</a>
          <a id="btnIdle" onClick={ this.onClickBtnIdle } className="top-menu-item">Stop</a>
        </TopMenu>

        <a-scene>

          <a-assets>
            <img id="sky" src="img/square.jpg"/>
            <a-asset-item id="botmodel" src="models/bot4.json"></a-asset-item>
          </a-assets>

          <a-sky src="#sky" rotation="0 -90 0"/>

          <a-entity position="0 1 0">
            <a-entity position="0 -0.5 1.25" camera="userHeight: 0; near: 0.5; fov: 100"
            orbit-controls={ AFRAME.utils.styleParser.stringify(this.orbitControls) }/>
          </a-entity>

          <a-entity id="bot" ref={ (bot: AFrame.Entity) => this.refs.bot = bot }
            scale="1 1 1"
            position="0 0 0"
            animation-mixer={ `clip: ${this.state.botStatus}; crossFadeDuration: ${this.crossFadeDuration}` }
            json-model="src: #botmodel">
          </a-entity>

          <a-plane height="100" width="33" color="#4d672b" rotation="-90 0 0"></a-plane>

        </a-scene>
      </div>
    )
  }
}