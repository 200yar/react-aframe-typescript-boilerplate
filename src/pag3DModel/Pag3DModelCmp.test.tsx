import 'aframe';
import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import {Pag3DModel} from "./Pag3DModel";

describe('Pag3DModel Component', () => {

  const component = shallow(<Pag3DModel>Dummy content</Pag3DModel>);

  it("is defined", () => {
    expect(component).toBeDefined();
  });

  it('renders without crashing', () => {
    expect(component).toBeInstanceOf(ShallowWrapper);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

});