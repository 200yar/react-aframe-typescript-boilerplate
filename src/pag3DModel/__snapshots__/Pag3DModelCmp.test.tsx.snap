// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pag3DModel Component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pag3DModel>
    Dummy content
</Pag3DModel>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Loader>
          Loading
</Loader>,
        <SideMenu
          itemActive="3"
          items={
                    Array [
                              Object {
                                "active": false,
                                "ico": "start.svg",
                                "name": "Home",
                                "url": "#/",
                              },
                              Object {
                                "active": false,
                                "ico": "video-player.svg",
                                "name": "2D/3D Video",
                                "url": "#/2dvideo",
                              },
                              Object {
                                "active": false,
                                "ico": "360-degrees.svg",
                                "name": "360 Video",
                                "url": "#/360video",
                              },
                              Object {
                                "active": false,
                                "ico": "3d-printer.svg",
                                "name": "3D Model Animation",
                                "url": "#/3dmodel",
                              },
                            ]
          }
          title="3D Model Animation"
/>,
        <TopMenu
          onClickMenuBtn={[Function]}
>
          <a
                    className="top-menu-item"
                    id="btnIdle"
                    onClick={[Function]}
          >
                    Walk
          </a>
          <a
                    className="top-menu-item"
                    id="btnRun"
                    onClick={[Function]}
          >
                    Run
          </a>
          <a
                    className="top-menu-item"
                    id="btnIdle"
                    onClick={[Function]}
          >
                    Jump
          </a>
          <a
                    className="top-menu-item"
                    id="btnIdle"
                    onClick={[Function]}
          >
                    Stop
          </a>
</TopMenu>,
        <a-scene>
          <a-assets>
                    <img
                              id="sky"
                              src="img/square.jpg"
                    />
                    <a-asset-item
                              id="botmodel"
                              src="models/bot4.json"
                    />
          </a-assets>
          <a-sky
                    rotation="0 -90 0"
                    src="#sky"
          />
          <a-entity
                    position="0 1 0"
          >
                    <a-entity
                              camera="userHeight: 0; near: 0.5; fov: 100"
                              orbit-controls="autoRotate:true;target:#bot;enableDamping:true;dampingFactor:0.1;rotateSpeed:0.1;autoRotateSpeed:0.15;zoomSpeed:0.5;minDistance:0;maxDistance:100;invertZoom:true"
                              position="0 -0.5 1.25"
                    />
          </a-entity>
          <a-entity
                    animation-mixer="clip: idle; crossFadeDuration: 0.5"
                    id="bot"
                    json-model="src: #botmodel"
                    position="0 0 0"
                    scale="1 1 1"
          />
          <a-plane
                    color="#4d672b"
                    height="100"
                    rotation="-90 0 0"
                    width="33"
          />
</a-scene>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Loading",
        },
        "ref": "loader",
        "rendered": "Loading",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "itemActive": "3",
          "items": Array [
            Object {
              "active": false,
              "ico": "start.svg",
              "name": "Home",
              "url": "#/",
            },
            Object {
              "active": false,
              "ico": "video-player.svg",
              "name": "2D/3D Video",
              "url": "#/2dvideo",
            },
            Object {
              "active": false,
              "ico": "360-degrees.svg",
              "name": "360 Video",
              "url": "#/360video",
            },
            Object {
              "active": false,
              "ico": "3d-printer.svg",
              "name": "3D Model Animation",
              "url": "#/3dmodel",
            },
          ],
          "title": "3D Model Animation",
        },
        "ref": "sideMenu",
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <a
              className="top-menu-item"
              id="btnIdle"
              onClick={[Function]}
>
              Walk
</a>,
            <a
              className="top-menu-item"
              id="btnRun"
              onClick={[Function]}
>
              Run
</a>,
            <a
              className="top-menu-item"
              id="btnIdle"
              onClick={[Function]}
>
              Jump
</a>,
            <a
              className="top-menu-item"
              id="btnIdle"
              onClick={[Function]}
>
              Stop
</a>,
          ],
          "onClickMenuBtn": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Walk",
              "className": "top-menu-item",
              "id": "btnIdle",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Walk",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Run",
              "className": "top-menu-item",
              "id": "btnRun",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Run",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Jump",
              "className": "top-menu-item",
              "id": "btnIdle",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Jump",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Stop",
              "className": "top-menu-item",
              "id": "btnIdle",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Stop",
            "type": "a",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a-assets>
              <img
                            id="sky"
                            src="img/square.jpg"
              />
              <a-asset-item
                            id="botmodel"
                            src="models/bot4.json"
              />
</a-assets>,
            <a-sky
              rotation="0 -90 0"
              src="#sky"
/>,
            <a-entity
              position="0 1 0"
>
              <a-entity
                            camera="userHeight: 0; near: 0.5; fov: 100"
                            orbit-controls="autoRotate:true;target:#bot;enableDamping:true;dampingFactor:0.1;rotateSpeed:0.1;autoRotateSpeed:0.15;zoomSpeed:0.5;minDistance:0;maxDistance:100;invertZoom:true"
                            position="0 -0.5 1.25"
              />
</a-entity>,
            <a-entity
              animation-mixer="clip: idle; crossFadeDuration: 0.5"
              id="bot"
              json-model="src: #botmodel"
              position="0 0 0"
              scale="1 1 1"
/>,
            <a-plane
              color="#4d672b"
              height="100"
              rotation="-90 0 0"
              width="33"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  id="sky"
                  src="img/square.jpg"
/>,
                <a-asset-item
                  id="botmodel"
                  src="models/bot4.json"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "sky",
                  "src": "img/square.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "botmodel",
                  "src": "models/bot4.json",
                },
                "ref": null,
                "rendered": null,
                "type": "a-asset-item",
              },
            ],
            "type": "a-assets",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "rotation": "0 -90 0",
              "src": "#sky",
            },
            "ref": null,
            "rendered": null,
            "type": "a-sky",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a-entity
                camera="userHeight: 0; near: 0.5; fov: 100"
                orbit-controls="autoRotate:true;target:#bot;enableDamping:true;dampingFactor:0.1;rotateSpeed:0.1;autoRotateSpeed:0.15;zoomSpeed:0.5;minDistance:0;maxDistance:100;invertZoom:true"
                position="0 -0.5 1.25"
/>,
              "position": "0 1 0",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "camera": "userHeight: 0; near: 0.5; fov: 100",
                "orbit-controls": "autoRotate:true;target:#bot;enableDamping:true;dampingFactor:0.1;rotateSpeed:0.1;autoRotateSpeed:0.15;zoomSpeed:0.5;minDistance:0;maxDistance:100;invertZoom:true",
                "position": "0 -0.5 1.25",
              },
              "ref": null,
              "rendered": null,
              "type": "a-entity",
            },
            "type": "a-entity",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "animation-mixer": "clip: idle; crossFadeDuration: 0.5",
              "id": "bot",
              "json-model": "src: #botmodel",
              "position": "0 0 0",
              "scale": "1 1 1",
            },
            "ref": [Function],
            "rendered": null,
            "type": "a-entity",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "color": "#4d672b",
              "height": "100",
              "rotation": "-90 0 0",
              "width": "33",
            },
            "ref": null,
            "rendered": null,
            "type": "a-plane",
          },
        ],
        "type": "a-scene",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Loader>
            Loading
</Loader>,
          <SideMenu
            itemActive="3"
            items={
                        Array [
                                    Object {
                                      "active": false,
                                      "ico": "start.svg",
                                      "name": "Home",
                                      "url": "#/",
                                    },
                                    Object {
                                      "active": false,
                                      "ico": "video-player.svg",
                                      "name": "2D/3D Video",
                                      "url": "#/2dvideo",
                                    },
                                    Object {
                                      "active": false,
                                      "ico": "360-degrees.svg",
                                      "name": "360 Video",
                                      "url": "#/360video",
                                    },
                                    Object {
                                      "active": false,
                                      "ico": "3d-printer.svg",
                                      "name": "3D Model Animation",
                                      "url": "#/3dmodel",
                                    },
                                  ]
            }
            title="3D Model Animation"
/>,
          <TopMenu
            onClickMenuBtn={[Function]}
>
            <a
                        className="top-menu-item"
                        id="btnIdle"
                        onClick={[Function]}
            >
                        Walk
            </a>
            <a
                        className="top-menu-item"
                        id="btnRun"
                        onClick={[Function]}
            >
                        Run
            </a>
            <a
                        className="top-menu-item"
                        id="btnIdle"
                        onClick={[Function]}
            >
                        Jump
            </a>
            <a
                        className="top-menu-item"
                        id="btnIdle"
                        onClick={[Function]}
            >
                        Stop
            </a>
</TopMenu>,
          <a-scene>
            <a-assets>
                        <img
                                    id="sky"
                                    src="img/square.jpg"
                        />
                        <a-asset-item
                                    id="botmodel"
                                    src="models/bot4.json"
                        />
            </a-assets>
            <a-sky
                        rotation="0 -90 0"
                        src="#sky"
            />
            <a-entity
                        position="0 1 0"
            >
                        <a-entity
                                    camera="userHeight: 0; near: 0.5; fov: 100"
                                    orbit-controls="autoRotate:true;target:#bot;enableDamping:true;dampingFactor:0.1;rotateSpeed:0.1;autoRotateSpeed:0.15;zoomSpeed:0.5;minDistance:0;maxDistance:100;invertZoom:true"
                                    position="0 -0.5 1.25"
                        />
            </a-entity>
            <a-entity
                        animation-mixer="clip: idle; crossFadeDuration: 0.5"
                        id="bot"
                        json-model="src: #botmodel"
                        position="0 0 0"
                        scale="1 1 1"
            />
            <a-plane
                        color="#4d672b"
                        height="100"
                        rotation="-90 0 0"
                        width="33"
            />
</a-scene>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Loading",
          },
          "ref": "loader",
          "rendered": "Loading",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "itemActive": "3",
            "items": Array [
              Object {
                "active": false,
                "ico": "start.svg",
                "name": "Home",
                "url": "#/",
              },
              Object {
                "active": false,
                "ico": "video-player.svg",
                "name": "2D/3D Video",
                "url": "#/2dvideo",
              },
              Object {
                "active": false,
                "ico": "360-degrees.svg",
                "name": "360 Video",
                "url": "#/360video",
              },
              Object {
                "active": false,
                "ico": "3d-printer.svg",
                "name": "3D Model Animation",
                "url": "#/3dmodel",
              },
            ],
            "title": "3D Model Animation",
          },
          "ref": "sideMenu",
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <a
                className="top-menu-item"
                id="btnIdle"
                onClick={[Function]}
>
                Walk
</a>,
              <a
                className="top-menu-item"
                id="btnRun"
                onClick={[Function]}
>
                Run
</a>,
              <a
                className="top-menu-item"
                id="btnIdle"
                onClick={[Function]}
>
                Jump
</a>,
              <a
                className="top-menu-item"
                id="btnIdle"
                onClick={[Function]}
>
                Stop
</a>,
            ],
            "onClickMenuBtn": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Walk",
                "className": "top-menu-item",
                "id": "btnIdle",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Walk",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Run",
                "className": "top-menu-item",
                "id": "btnRun",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Run",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Jump",
                "className": "top-menu-item",
                "id": "btnIdle",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Jump",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Stop",
                "className": "top-menu-item",
                "id": "btnIdle",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Stop",
              "type": "a",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a-assets>
                <img
                                id="sky"
                                src="img/square.jpg"
                />
                <a-asset-item
                                id="botmodel"
                                src="models/bot4.json"
                />
</a-assets>,
              <a-sky
                rotation="0 -90 0"
                src="#sky"
/>,
              <a-entity
                position="0 1 0"
>
                <a-entity
                                camera="userHeight: 0; near: 0.5; fov: 100"
                                orbit-controls="autoRotate:true;target:#bot;enableDamping:true;dampingFactor:0.1;rotateSpeed:0.1;autoRotateSpeed:0.15;zoomSpeed:0.5;minDistance:0;maxDistance:100;invertZoom:true"
                                position="0 -0.5 1.25"
                />
</a-entity>,
              <a-entity
                animation-mixer="clip: idle; crossFadeDuration: 0.5"
                id="bot"
                json-model="src: #botmodel"
                position="0 0 0"
                scale="1 1 1"
/>,
              <a-plane
                color="#4d672b"
                height="100"
                rotation="-90 0 0"
                width="33"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    id="sky"
                    src="img/square.jpg"
/>,
                  <a-asset-item
                    id="botmodel"
                    src="models/bot4.json"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "sky",
                    "src": "img/square.jpg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "botmodel",
                    "src": "models/bot4.json",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "a-asset-item",
                },
              ],
              "type": "a-assets",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "rotation": "0 -90 0",
                "src": "#sky",
              },
              "ref": null,
              "rendered": null,
              "type": "a-sky",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a-entity
                  camera="userHeight: 0; near: 0.5; fov: 100"
                  orbit-controls="autoRotate:true;target:#bot;enableDamping:true;dampingFactor:0.1;rotateSpeed:0.1;autoRotateSpeed:0.15;zoomSpeed:0.5;minDistance:0;maxDistance:100;invertZoom:true"
                  position="0 -0.5 1.25"
/>,
                "position": "0 1 0",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "camera": "userHeight: 0; near: 0.5; fov: 100",
                  "orbit-controls": "autoRotate:true;target:#bot;enableDamping:true;dampingFactor:0.1;rotateSpeed:0.1;autoRotateSpeed:0.15;zoomSpeed:0.5;minDistance:0;maxDistance:100;invertZoom:true",
                  "position": "0 -0.5 1.25",
                },
                "ref": null,
                "rendered": null,
                "type": "a-entity",
              },
              "type": "a-entity",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "animation-mixer": "clip: idle; crossFadeDuration: 0.5",
                "id": "bot",
                "json-model": "src: #botmodel",
                "position": "0 0 0",
                "scale": "1 1 1",
              },
              "ref": [Function],
              "rendered": null,
              "type": "a-entity",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "color": "#4d672b",
                "height": "100",
                "rotation": "-90 0 0",
                "width": "33",
              },
              "ref": null,
              "rendered": null,
              "type": "a-plane",
            },
          ],
          "type": "a-scene",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
